cw_ethereum/lib/ethereum_wallet.dart

```diff
@@ -1,126 +1,59 @@
-import 'dart:async';
 import 'dart:convert';
-import 'dart:io';
-import 'dart:math';
 
-import 'package:cw_core/crypto_currency.dart';
 import 'package:cw_core/cake_hive.dart';
-import 'package:cw_core/node.dart';
+import 'package:cw_core/crypto_currency.dart';
+import 'package:cw_core/currency_for_wallet_type.dart';
+import 'package:cw_core/erc20_token.dart';
 import 'package:cw_core/pathForWallet.dart';
-import 'package:cw_core/pending_transaction.dart';
-import 'package:cw_core/sync_status.dart';
 import 'package:cw_core/transaction_direction.dart';
-import 'package:cw_core/transaction_priority.dart';
-import 'package:cw_core/wallet_addresses.dart';
-import 'package:cw_core/wallet_base.dart';
 import 'package:cw_core/wallet_info.dart';
 import 'package:cw_ethereum/default_ethereum_erc20_tokens.dart';
-import 'package:cw_ethereum/erc20_balance.dart';
 import 'package:cw_ethereum/ethereum_client.dart';
-import 'package:cw_ethereum/ethereum_exceptions.dart';
-import 'package:cw_ethereum/ethereum_formatter.dart';
-import 'package:cw_ethereum/ethereum_transaction_credentials.dart';
 import 'package:cw_ethereum/ethereum_transaction_history.dart';
-import 'package:cw_ethereum/ethereum_transaction_info.dart';
-import 'package:cw_ethereum/ethereum_transaction_model.dart';
-import 'package:cw_ethereum/ethereum_transaction_priority.dart';
-import 'package:cw_ethereum/ethereum_wallet_addresses.dart';
-import 'package:cw_ethereum/file.dart';
-import 'package:cw_core/erc20_token.dart';
-import 'package:hive/hive.dart';
-import 'package:hex/hex.dart';
-import 'package:mobx/mobx.dart';
-import 'package:shared_preferences/shared_preferences.dart';
-import 'package:web3dart/crypto.dart';
-import 'package:web3dart/web3dart.dart';
-import 'package:bip39/bip39.dart' as bip39;
-import 'package:bip32/bip32.dart' as bip32;
-
-part 'ethereum_wallet.g.dart';
+import 'package:cw_evm/evm_chain_transaction_history.dart';
+import 'package:cw_evm/evm_chain_transaction_info.dart';
+import 'package:cw_evm/evm_chain_transaction_model.dart';
+import 'package:cw_evm/evm_chain_wallet.dart';
+import 'package:cw_evm/evm_erc20_balance.dart';
+import 'package:cw_evm/file.dart';
+
+class EthereumWallet extends EVMChainWallet {
+  EthereumWallet({
+    required super.client,
+    required super.password,
+    required super.walletInfo,
+    super.mnemonic,
+    super.initialBalance,
+    super.privateKey,
+    super.nativeCurrency = CryptoCurrency.eth,
```

Comment:
we are 100% sure that this should be eth, so no need to take it as a parameter and just send it to the `super` call directly as eth, and for matic polygon as well, no need for the extra `getCurrecyForWalletType` in wallet_service